src/components/JobWizard/JobWizardSteps/JobWizardFunctionSelection/JobWizardFunctionSelection.js

```diff
@@ -230,72 +229,80 @@ const JobWizardFunctionSelection = ({
+        }, {})
+      )
 
-        setFunctions(groupedFunctions)
+      setFunctions(groupedFunctions)
 
-        if (filterByName.length > 0) {
-          const filteredFunctions = validFunctions.filter(func => {
-            return func.metadata.name.includes(filterByName)
-          })
+      if (filterByName.length > 0) {
+        const filteredFunctions = validFunctions.filter(func => {
+          return func.metadata.name.includes(filterByName)
+        })
 
-          setFilteredFunctions(filteredFunctions)
-        }
-      })
-    }
-    const fetchHubFunctionsPromise = () => {
-      return fetchHubFunctions().then(templatesObject => {
-        if (templatesObject) {
-          setTemplatesCategories(templatesObject.hubFunctionsCategories)
-          setTemplates(templatesObject.hubFunctions)
-
-          formState.initialValues.functionSelection.templatesLabels =
-            templatesObject.hubFunctions.reduce((labels, template) => {
-              labels[template.metadata.name] = template.ui.categories.map(categoryId => {
-                return {
-                  id: categoryId,
-                  key: getCategoryName(categoryId),
-                  isKeyOnly: true
-                }
-              })
-
-              return labels
-            }, {})
-        }
-      })
-    }
-    const promisesArray = [fetchFunctionsPromise]
+        setFilteredFunctions(filteredFunctions)
+      }
+    })
 
-    setIsLoading(true)
     formState.initialValues.functionSelection.projectName = currentValue
+  }
 
-    if (isEmpty(templatesCategories) || isEmpty(templates)) {
-      promisesArray.push(fetchHubFunctionsPromise)
-    }
+  const fetchHubFunctionsPromise = useCallback(() => {
+    fetchHubFunctions().then(templatesObject => {
+      if (templatesObject) {
+        setTemplatesCategories(templatesObject.hubFunctionsCategories)
+        setTemplates(templatesObject.hubFunctions)
+
+        formState.initialValues.functionSelection.templatesLabels =
+          templatesObject.hubFunctions.reduce((labels, template) => {
+            labels[template.metadata.name] = template.ui.categories.map(categoryId => {
+              return {
+                id: categoryId,
+                key: getCategoryName(categoryId),
+                isKeyOnly: true
+              }
+            })
 
-    Promise.all([promisesArray.map(promise => promise())]).finally(() => {
-      setIsLoading(false)
+            return labels
+          }, {})
+      }
     })
-  }
+  }, [
+    fetchHubFunctions,
+    formState.initialValues.functionSelection,
+    setTemplates,
+    setTemplatesCategories
+  ])
+
+  useEffect(() => {
+    if (
+      activeTab === FUNCTIONS_SELECTION_HUB_TAB &&
+      (isEmpty(functionsStore.hubFunctions) || isEmpty(functionsStore.hubFunctionsCatalog))
+    ) {
+      fetchHubFunctionsPromise()
+    }
+  }, [
+    activeTab,
+    functionsStore.hubFunctions,
+    functionsStore.hubFunctionsCatalog,
+    fetchHubFunctionsPromise
+  ])
```

Comment:
There is no need to have a separate `fetchHubFunctionsPromise` function. 
Also, It's better not to pass the entire functionsStore to this components. Probably you can use `templates` and `templatesCategories` instead